src/test/java/org/mitre/dsmiley/httpproxy/ChunkedTransferTest.java

```diff
@@ -123,6 +123,73 @@ protected void doGet(HttpServletRequest req, HttpServletResponse resp) throws Se
     }
   }
 
+  @Test
+  public void testChunkedTransferClosing() throws Exception {
+    /*
+     This test ensures, that the chunk encoded backing connection is closed,
+     when the closing of the proxy (frontend) connection is detected.
+
+     The idea is, that in the servlet the closing of the backing connection can
+     be detected, because the output stream is closed and an IOException is
+     raised. If no exception is raised when writing multiple chunks, it must
+     be assumed, that the backing connection is not closed.
+     */
+    final CountDownLatch guardForSecondRead = new CountDownLatch(1);
+    final CountDownLatch guardForEnd = new CountDownLatch(1);
+    final byte[] data1 = "event: message\ndata: Dummy Data1\n\n".getBytes(StandardCharsets.UTF_8);
+    final byte[] data2 = "event: message\ndata: Dummy Data2\n\n".getBytes(StandardCharsets.UTF_8);
```

Comment:
not used

Reply:
Ah - yes, I reworked that test, which rendered the third data sample obsolete.